internal/service/lightsail/database.go

```diff
@@ -0,0 +1,485 @@
+	}
+
+	// Backup Retention is not a value you can pass on creation and defaults to true.
+	// Forcing an update of the value after creation if the backup_retention_enabled value is false.
+	if v := d.Get("backup_retention_enabled"); v == false {
+		log.Printf("[DEBUG] Lightsail Database (%s) backup_retention_enabled setting is false. Updating value.", d.Id())
+		req := lightsail.UpdateRelationalDatabaseInput{
+			ApplyImmediately:       aws.Bool(true),
+			RelationalDatabaseName: aws.String(d.Id()),
+			DisableBackupRetention: aws.Bool(true),
+		}
+
+		resp, err := conn.UpdateRelationalDatabase(&req)
+		if err != nil {
+			return err
+		}
+
+		if len(resp.Operations) == 0 {
+			return fmt.Errorf("No operations found for Update Relational Database request")
+		}
+
+		op := resp.Operations[0]
+
+		err = waitLightsailOperation(conn, op.Id)
+		if err != nil {
+			return fmt.Errorf("Error waiting for Relational Database (%s) to become ready: %s", d.Id(), err)
+		}
+
+		err = waitDatabaseBackupRetentionModified(conn, aws.String(d.Id()), aws.Bool(v.(bool)))
+		if err != nil {
+			return fmt.Errorf("Error waiting for Relational Database (%s) Backup Retention to be updated: %s", d.Id(), err)
+		}
+
+	}
+
+	// Some Operations can complete before the Database enters the Available state. Added a waiter to make sure the Database is available before continuing.
+	err = waitDatabaseModified(conn, aws.String(d.Id()))
+	if err != nil {
+		return fmt.Errorf("Error waiting for Relational Database (%s) to become available: %s", d.Id(), err)
+	}
+
+	return ResourceDatabaseRead(d, meta)
+}
+
+func ResourceDatabaseRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).LightsailConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	// Some Operations can complete before the Database enters the Available state. Added a waiter to make sure the Database is available before continuing.
+	// This is to support importing a resource that is not in a ready state.
+	err := waitDatabaseModified(conn, aws.String(d.Id()))
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "NotFoundException" {
+				log.Printf("[WARN] Lightsail Relational Database (%s) not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+		}
+		return fmt.Errorf("Error waiting for Relational Database (%s) to become available: %s", d.Id(), err)
+	}
+
+	resp, err := conn.GetRelationalDatabase(&lightsail.GetRelationalDatabaseInput{
+		RelationalDatabaseName: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "NotFoundException" {
+				log.Printf("[WARN] Lightsail Relational Database (%s) not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return err
+		}
+		return err
+	}
```

Comment:
maybe to reduce the logic here, ` waitDatabaseModified` can also return the database API object since within that method we call `GetRelationalDatabase`

```suggestion
	database, err := waitDatabaseModified(conn, aws.String(d.Id()))
	
	if !d.IsNewResource() && tfresource.NotFound(err){
            log.Printf("[WARN] Lightsail Relational Database (%s) not found, removing from state", d.Id())
            d.SetId("")
            return nil
	}
	
	if err != nil {
		return fmt.Errorf("error reading LightSail Relational Database (%s): %w", d.Id(), err)
	}
```

Reply:
Updated waiter to return the database object and removed the duplicate logic in the Read operation.